### ISSUE TITLE
Changes to Instagram API policy has broken Instafeed

### ISSUE THREAD

**Issue by TamaEaston** [stevenschobert/instafeed.js#408]
Instafeed is now returning the following error: 

**Uncaught Error: Error from Instagram: The access_token provided does not match an approved application.**

Instagram have changed their API policy, effective as of the 1st of June: https://www.instagram.com/developer/endpoints/users/

> **Instagram Platform and documentation update.** Apps created on or after Nov 17, 2015 will start in Sandbox Mode and function on newly updated API rate-limits and behaviors. Prior to going Live, and being able to be used by people other than the developers of the app, these apps will have to go through a new review process. Please read the API documentation or the Change Log for more details.
> 
> Any app created before Nov 17, 2015 will continue to function **until June 1, 2016**. On that date, the app will automatically be moved to Sandbox Mode if it wasn't approved through the review process. The previous version of our documentation is still available here.

Is development for Instafeed going to continue to address these changes?

**Comment by rhcarlos** [comment]
I hope so
I need this change, i have a lot clients using instafeed.js, and all is broken now :disappointed:

**Comment by mikeden** [comment]
Same boat here - just launched a website yesterday for a client that uses instafeed and now it's broken.  Help!

**Comment by benrellick** [comment]
I think I've found a way around this. It's not ideal, but it works. You'll have to have your client go to http://instagram.pixelunion.net/ (while they're logged in to Instagram) and generate an access token. Add that to your instafeed code along with the userId and clientId and that should work.

From the documentation:
`var userFeed = new Instafeed({
  get: 'user',
  userId: 'YOUR_USER_ID',
  accessToken: 'YOUR_ACCESS_TOKEN'
});
userFeed.run();`

**Comment by rhcarlos** [comment]
here don't work =\ show error message ```Uncaught SyntaxError: Unexpected token ;```

**Comment by thomaskimura** [comment]
@benrellick Worked for me. Thankfully this is exactly what I was using instafeed for in the first place.

**Comment by mikeden** [comment]
I now get a different error: 'This client has not been approved to access this resource.'

**Comment by rachelrianne** [comment]
@benrellick — that's great! Works for user feeds, but **doesn't work if you're trying to use 'tag' for the 'get' option.** 

For tagged images, you have to go through some process to move from Sandbox mode in order to access public_content, and if it's just a website, Insta doesn't even let you go through the process from what I've found so far.

**Comment by bluesix** [comment]
Bottom line is you will need to generate a new Access Token. Do that at the pixelunion link or http://instagramwordpress.rafsegat.com/docs/get-access-token/.
And you can only do "user:get" now from the looks of things.

**Comment by jibroin** [comment]
@bluesix thats good. However, is it can be applied to the case is not a word press?

**Comment by bluesix** [comment]
@jibroin are you asking if it works for non-Wordpress sites? I've just fixed 4 clients sites - 2 Wordpress, 2 hand-coded. All good.

**Comment by virajsoni06** [comment]
Hi @bluesix I wanted to know how were you able get your app out of sandbox mode. 

When I start a submission and select "I want to display my Instagram posts on my website." it just shows me a notification box. I couldn't find a way to get it out of Sandbox mode

Thanks

**Comment by thefunkybassist** [comment]
As far as I understand it, it shouldn't be neccesary to get out of Sandbox for public_content level feeds? At least, I can get it to authorize for that scope by adding it to the access token generator like this:
https://www.instagram.com/oauth/authorize/?client_id=...&redirect_uri=...&response_type=token&scope=public_content
To prevent double posts, see my other topic response [here](https://github.com/stevenschobert/instafeed.js/issues/345#issuecomment-223223289).

No authentication errors after that. But I can't seem get any "tagged" content, just empty response from Instagram.

**Comment by bluesix** [comment]
@virajsoni06 What @thefunkybassist said. As long as you're just doing a `get: user` (i.e. just displaying a user's feed) then you don't need to do the app review thing, just need to generate a new access token.

**Comment by thefunkybassist** [comment]
Does anybody know how to get a working access token for "I want to display hashtag content and public content on my website.", aka tagged content? It says Instagram doesn't support this use case? Confusing.

**Comment by bogdanripa** [comment]
This access_token solution does not sound like a secure option though...

**Comment by jonnyjaniero** [comment]
I'm using it to pull tags on product page of woocommerce site.
Generating a new token and including it seemed to fixed my problem

Token:
http://instagramwordpress.rafsegat.com/docs/get-access-token/

`
    var feed = new Instafeed({
        get: 'tagged',
        tagName: 'YOUR_TAG_NAME',
        userId: 'YOUR_USER_ID',
        accessToken: 'YOUR_ACCESS_TOKEN',
        sortBy: 'most-recent',
        limit: '100'
    });
    feed.run();
`

**Comment by Rubicrash** [comment]
@jonnyjaniero Fantastic! It worked for me. Thank you very much

**Comment by leo07vasp** [comment]
@jonnyjaniero how work this ? get pictures with tag ?

**Comment by jehzlau** [comment]
Same here. It stopped working just 24 hours ago. I thought I'm alone, good thing I'm not. :D

**Comment by safarious** [comment]
Is there some one already working on a fix for this?

**Comment by jehzlau** [comment]
@benrellick awesome it works. Thanks for sharing! Yay!

I used this one to generate my access token before: http://www.pinceladasdaweb.com.br/instagram/access-token/

Currently, it doesn't work. Maybe the app by pinceladasdaweb.com.br needs to be approved first, because I assume it's currently in Sandbox Mode. :D

So for those who encountered the same problem, let's just generate a new access token via http://instagram.pixelunion.net/

**Comment by jehzlau** [comment]
@safarious - yes you can generate a new access token here http://instagram.pixelunion.net/

**Comment by safarious** [comment]
@jehzlau so I need to add to properties now?:
	```
		accessToken: 'XXXX',
			    clientId: 'XXXX,
```

**Comment by jehzlau** [comment]
@safarious there's an accesstoken in clientID before. You just need to replace it. :)

**Comment by safarious** [comment]
@jehzlau thank you for your fast response - where do I get the userId ?

**Comment by undergroundmedia** [comment]
The user ID is very simple as it can be seen within the Access Tokens.
It will be the first 8-9 charcters before the .  for example:

12345678.abcd4567789

**Comment by JoshHiggins** [comment]
The http://instagram.pixelunion.net/ token works for user feeds, but not hashtag or location based feeds. Anyone have a line on a place to get these feeds?

**Comment by thefunkybassist** [comment]
@JoshHiggins apparently Instagram changed their API functionality so now you can't get public hashtag feeds. Only your own user feed. Ridiculous but true!

**Comment by undergroundmedia** [comment]
Tags are a problem but they do present an intellectual property issue in that you are displaying what you do not have the rights to broadcast so there should be strict control of that.

**Comment by cesarvelame** [comment]
I'm getting error 429, too many requests. 

`Failed to load resource: the server responded with a status of 429 ()`

**Comment by safarious** [comment]
@undergroundmedia @jehzlau @jonnyjaniero 
Hey Guys I've just added the userId & accessToken but I do get following error: `Error from Instagram: This client has not been approved to access this resource.` (I've got the similar implementation as you do @jonnyjaniero )

Do someone know a fix for this?

**Comment by undergroundmedia** [comment]
@safarious Where did you get the Access token from?

**Comment by safarious** [comment]
@undergroundmedia  I've got it from here: http://instagram.pixelunion.net/ but it seems that they do not work for that usecase if I take the accesToken from here: http://instagramwordpress.rafsegat.com/docs/get-access-token/ it works.

**Comment by undergroundmedia** [comment]
Could be Pixel Unions client having issues from Instagram?
We are still waiting for them to pull their finger out and restore ours to live instead of Sandbox which is causing a lot of issues here for our support guys.

Half tempted to throw out an instruction doc on how to create a client and generate your own tokens for Sandbox only just to show your own images on your own site but it is a lot to ask for non coding users to have to go through.

**Comment by codytooker** [comment]
So before I was using this to simply pull in my clients instagram feeds to their site.  I was able to pull in anyone's feed with my secret key.  Now it seems like you are only allowed to pull in your own instagram images is this correct?

So every client needs to generate a new access token in order for this to work again.   Thats kind of a bummer

**Comment by djrasmusp** [comment]
Anybody that knows of where i can get a Access Token, that does have the "non-square"-setting enabled ?

**Comment by rorysmorris** [comment]
You will not need every client to generate a new access token. Creating an access token here, from your own account will fix the issue: http://instagramwordpress.rafsegat.com/docs/get-access-token/ - PixelUnion doesn't work.

**Comment by yiyonglee** [comment]
thanks @rorysmorris, works for my client

(1) Go to http://instagramwordpress.rafsegat.com/docs/get-access-token/
(2) Bottom of the page, click "Click here to get your Acess Token!"
(3) In your javascript, add **accessToken: '_YOUR_NEWLY_GENERATED_ACCESS_TOKEN_'**

My feed pull via "tagName" and it's working: http://www.tesco.com.my/havoc/

**Comment by undergroundmedia** [comment]
@TamaEaston  We have our app approved and live so if you are trying to gain the Access tokens for user only feeds in your Shopify themes here is the URL for that: [http://www.shopifythemes.net/pages/instagram/](url)

**Comment by benrellick** [comment]
Nice, thanks @rorysmorris!

The access token generated at http://instagramwordpress.rafsegat.com/docs/get-access-token/ grants permission for both the **basic** and **public_content** scope.

**Comment by BMCwebdev** [comment]
Any way to generate an access token without giving access to these other apps?

**Comment by crnr** [comment]
Is there any possibility to get specific hashtag's within this newest API changes ?

Thanks guys

**Comment by Rubicrash** [comment]
For hashtags: get a new access token from http://instagramwordpress.rafsegat.com/docs/get-access-token/ (or http://instagram.pixelunion.net/ , ..)

 var feed = new Instafeed({ 
	get: 'tagged', 
	tagName: '{{ hastag }}', 
	userId: '{{ your-user-id }}', 
	accessToken: '{{ access-token}}', 
}); 
feed.run();

**Comment by droozh** [comment]
You can get access to user feed who gave you access to his data. Before 01.06.2016 you could get data of any users feed while you in a sandbox. 
That's the problem. Everything changed. Now you should parse needed information without api.

**Comment by ronilaukkarinen** [comment]
@jonnyjaniero Thanks a lot, you're saved my day! I was able to generate accessTokens for my own apps, but they don't have sufficient permissions to display public data and new permissions can't be applied any longer as Instagram forbids it. Your trick obviously generates accessToken with public_content rights. Working great. Many, many thanks.

**Comment by iamdave210** [comment]
I can't get an access_token. I went to the URL in the docs and it had a button to authorize along with a message about how only sandbox users can authorize. I clicked it the button and it sent me to a 404. Now the URL that originally gave me the authorize screen is just a 404 too. Frustrating. :(

**Comment by PandaStyle** [comment]
For how long are these generated access tokens going to last? The official docs says you shouldn't count with them forever.

And also if a large amount of poeple are going to use access tokens generated form the same clientID (in this case Wordpress or Shopify), isn't the hourly request amount going to reach the limit (5000) quickly?

**Comment by lozozzo92** [comment]
it's still broken for me

**Comment by yomguy** [comment]
http://instagramwordpress.rafsegat.com/docs/get-access-token/ worked fo me but not http://instagram.pixelunion.net/

**Comment by Hurmeli** [comment]
Does Instafeed work at all anymore? I'm trying to get my own feed via

`get: 'user',
 userdId: 'xxx',
 accessToken: 'xxx',
sortBy: 'most-recent',
 clientId: 'xxx',`

But nothing comes out. Token is new, in fact the whole app is new and it has permissions. Being a simple graphic designer, I'm unfortunately out of my depth with this. :'(

**Comment by M1chaelc** [comment]
I'm also getting error 429, too many requests.
Failed to load resource: the server responded with a status of 429 ()

My app is in sandbox mode and therefore limited to 500 requests per hour.
Is there an easy way to cache the results or alternatively get it out of sandbox?

**Comment by marcoj** [comment]
http://instagram.pixelunion.net/ doesn't work because they don't have permissions to public_content.
http://instagramwordpress.rafsegat.com/docs/get-access-token/ worked, but is it safe?

Anyway I don't think this is the proper procedure, can you recommend any other procedure?

Many thanks

**Comment by ronilaukkarinen** [comment]
It seems currently the only way is to use some accessToken that provides proper public_content permissions, like insatgramwordpress.rafsegat.com's, but I was wondering the same thing - if everyone uses it, how long it will work? watching this issue thread until someone figures out a better way...

**Comment by DaveTheGeek** [comment]
It worked for me using ClientID, UserID and AccessToken when i'm using the Tag option.

**Comment by Pelek7** [comment]
http://instagramwordpress.rafsegat.com/docs/get-access-token/ after one day of working it stops

> {"meta": {"error_type": "OAuthPermissionsException", "code": 400, "error_message": "This client has not been approved to access this resource."}}
>

Does anyone of you have the same ? should I by this control, or mabye someone figured out some better solution ?

**Comment by markhillard** [comment]
Registered my own client on Instagram and got my `accessToken` from the API. Pasted it into my Instafeed code and went back to the website I'm working on. Got this error in console:

```
Error: Error from Instagram: This request requires scope=public_content, but this access token is not authorized with this scope. The user must re-authorize your application with scope=public_content to be granted this permissions.
```

Checked client again and edited it. Under the "Permissions Review" area I see the following:

Which use case best describes your Instagram integration?

* I want to display hashtag content and public content on my website.

_This use case is not supported. We do not approve the public_content permission for one-off projects such as displaying hashtag based content on your website. As alternative solution, you can show your own Instagram content, or find a company that offers this type of service (content discover, moderation, and display)._

**WTF Instagram!?**

**Comment by JeremieForge** [comment]
I made a hand-coded app displaying a hashtag feed, it worked with an access token generated from elfsight (used on just 1 pc in localhost in a museum). Around june 1st the access token stopped working.

Yesterday I got an access token from rafsegat, it worked fine but it stopped working after one day.

Any other access token generated from pixel union, elfsight, etc. doesn't work..

Instagram clearly says I have to "find a company that offers this type of service (content discover, moderation, and display)" but none of these seem to work anymore !

**Comment by rorysmorris** [comment]
Me again!

The access token I generated at http://instagramwordpress.rafsegat.com/docs/get-access-token/  has now stopped working and once again my live sites are broken. It seems now pixelunion works again - but for how long?

**Comment by MTphilclothier** [comment]
This is a nightmare!! Every time we tell our clients to generate a new Access Token at pixel union, raf or elfsight (which is embarrassing in the first place) and then they stop working after a few days.

We've applied for access to the API twice and failed both times. We sell WordPress themes with custom Instagram feed integration via instafeed.js, but apparently this is not a supported use case.  Are they basically saying we need to use a third party WordPress plugin instead?

What the hell are they hoping to achieve with these new changes??

**Comment by rorysmorris** [comment]
@MTphilclothier a third party WordPress plugin probably will be facing the same issues. 

Having looked into this issue, the jist of the story is that Instagram no longer want you to be able to access other users content through their API. Accessing your own content is fine.

You can register an app to have the public_content scope; but you need to prove that it'll be doing one of the following, accompanied with a video screencast:

-  My app allows people to login with Instagram and share their own content.
-  My product helps brands and advertisers understand, manage their audience and media rights.
-  My product helps broadcasters and publishers discover content, get digital rights to media, and share media with proper attribution.

I'd imagine this is what PixelUnion/Rafsegat have done, but in actual fact their apps aren't doing any of those things, which probably answers why this problem has cropped up again. I would say using either of those services to generate an access token will be nothing more than a temporary fix. 

The most obvious, but painstaking solution is to gather the username and password for every one of my clients Instagram accounts. That's 50 - 100 accounts. 

Another possible solution would be to read the JS object displayed on Instagram's public profile pages. For instance, look at the source on https://www.instagram.com/instagram/ - there's a big data array at the bottom which contains everything we'd need to display recent posts from a particular user - we wouldn't need the user id or access token either - just do an AJAX load of the html of the page and grab the parts we need.

The problem with that, is that it'd only be a temporary solution - as soon as Instagram change the HTML layout of their page, it's broken again. 

Anyway, thanks Instagram. 👎

**Comment by MTphilclothier** [comment]
@rorysmorris yeah we have considered scraping the content via cURL instead, but this would breach Instagram's terms of service. It's so ironic that their actions are pushing developers to this sort of thinking though.

I guess we'll just have to keep working on "damage control" until we can figure out the magic words Instagram need in an application for the API. I might even email pixel union and raf to ask for their secret recipe. Luckily I don't think we compete with them directly.

**Comment by rorysmorris** [comment]
@MTphilclothier my colleague just sent me the video raf used to get his app approved - I just don't understand how this would be approved? https://www.youtube.com/watch?v=CfRua7XpJbo

**Comment by MTphilclothier** [comment]
@rorysmorris Wow, I guess there is no secret recipe then! haha. I'm going to re-submit our request. I'm guessing they must have a few data entry level workers going through the applications. Maybe it is just luck if you get a good/bad one?

**Comment by cesarvelame** [comment]
@M1chaelc I have the same problem: error 429 too many requests. I'm trying to figure out how cache the instagram feed.

**Comment by therofree** [comment]
Doesnt work anymore with token generated externaly. In the end i had to create an app, because i use it to get posts from my account doesnt need to be reviewed. I got my token from https://api.instagram.com/oauth/authorize/?client_id=CLIENT-ID&redirect_uri=REDIRECT-URI&response_type=token and works.

**Comment by apokyro** [comment]
Same problem here with tags.

I am trying to get images from my personal account but I don't know how to find user ID. All services that provide it are down. Can someone help?

Thanks in advance

**Comment by therofree** [comment]
Tags was what i also needed:
Create your own app and get the token from here https://api.instagram.com/oauth/authorize/?client_id=CLIENT-ID&redirect_uri=REDIRECT-URI&response_type=token

Then
<script type="text/javascript">
    var feed = new Instafeed({
        get: 'tagged',
        tagName: 'your_tag',
        accessToken: 'token'
    });
    feed.run();
</script>

**Comment by apokyro** [comment]
Thanks @therofree! It works with tags.

**Comment by Bolandish** [comment]
Maybe this can help you resolve your issue :)
https://github.com/Bolandish/Instagram-Hashtag-API

**Comment by markhillard** [comment]
@therofree and @apokyro, I did that very same thing and received the following error from the API:

```
This request requires scope=public_content, but this access token is not authorized with this scope. The user must re-authorize your application with scope=public_content to be granted this permissions.
```

The problem is that Instagram is not granting me `scope=public_content` for tags. With Instafeed, how did you get it to work with tags?

**Comment by ckeilitz** [comment]
Many of you have already posted this...but with the new Instagram policy user feeds is the only option and @benrellick's earlier post lays out the most simple user feed example. @thefunkybassist said it well, "Instagram changed their API functionality so now you can't get public hashtag feeds. Only your own user feed. Ridiculous but true!"  If you build in pagination, you can at least access all your user posts.

**Comment by markhillard** [comment]
This helped me out - http://stackoverflow.com/questions/33863505/oauthpermissionsexception-instagram-api-in-sandbox - but now my feed is only pulling photos that I took with the tag I set for Instafeed. Oh well, worth a shot I guess.

**Comment by therofree** [comment]
Cant confirm now but i remember i also unchecked Disable implicit OAuth.

**Comment by bloqhead** [comment]
@therofree were you able to successfully grab a user feed, or were you querying via something else like hashtag? I'm trying to get a simple user feed going and am even that is giving me problems.

**Comment by therofree** [comment]
I needed specific tags.

**Comment by tgirardi** [comment]
As well as @bogdanripa, I'm worried about the security implications of making the access_token publicly readable as it will occur with @benrellick's solution. Should I be worried about this or there is no danger?

**Comment by mtanaka14** [comment]
@Bolandish that will just get your domain banned by them.

**Comment by ckeilitz** [comment]
If anyone has an instafeed hastag feed working and is willing to share, please post it. I'm pretty sure there is no way to pull tags any longer...unless your app is approved - which seems nearly impossible since the approved use cases are so uncommon, particularly for normal instafeed users. If  the user feed working and substitute it with the tag feed, you'll get the following error:  Error from Instagram: This client has not been approved to access this resource.

**Comment by therofree** [comment]
@ckeilitz hashtags for your instagram account?

**Comment by ckeilitz** [comment]
You can definitely pull from your own instagram account w/o any special permission - something I have working just fine here:  http://www.hairxanadu.com/instagram.php

However, say for instance you wanted to pull a popular Instagram tag like #nike.  As of June 1st, 2016, this would fail with the error above - not approved to access this resource.  

So there is the challenge, I would love to see...get #nike working with instafeed.js.

**Comment by brucenorton** [comment]
Sharing what finally worked for me:
```
var userFeed = new Instafeed({
  get: 'user',
  userId: '710765',
  accessToken: '710765.REST-OF-ACCESS-TOKEN-GOES-HERE'
});
userFeed.run();
```

**Comment by Fanetteg** [comment]
hello, posting here after opening another issue -

got my new access token via pixelunion but it seems that it is not for non-square media (I only use vertical images - displaying my last instagram on my website) - also as I saw on this thread, it doesn't seem very reliable.

Could someone help with that ?

**Comment by Bolandish** [comment]
@mtanaka14 i don't care. :)
People can use it or dont use it :)

**Comment by bluesix** [comment]
Damn - all the feeds I fixed for various clients last week are dead again (token generated @ http://instagramwordpress.rafsegat.com/docs/get-access-token/) - getting the `Uncaught Error: Error from Instagram: The access_token provided does not match an approved application.` error.

**Comment by Fanetteg** [comment]
@bluesix answering my own issue here, generated my own token following these easy instructions : http://jelled.com/instagram/access-token and everything works perfectly now (even my issue with non-square format)

**Comment by uhlalars** [comment]
@Fanetteg I can't get this to work when trying to get images with certain hashtags. Have you tried that?

**Comment by ronilaukkarinen** [comment]
People need to read further up this thread. Please get it already: It is confirmed that **Instagram will NOT allow to display hashtag feeds after 1st of June via their API**. We are creating a scraper WordPress plugin just like @Bolandish's [here](https://github.com/stevenschobert/instafeed.js/issues/408#issuecomment-224282075). Ridiculous move from Instagram, but no can do.

**Comment by Fanetteg** [comment]
@uhlalars as @ronilaukkarinen said it doesn't work for hashtags ! But it's a solution to display an instagram feed - in my case my last instagram image

**Comment by uhlalars** [comment]
@ronilaukkarinen Yes, I read that. But people kept saying that they got it to work. Plus I got it to work myself a few days ago. But now it's completely undoable. Stupid, instagram.

**Comment by therofree** [comment]
It works and this is the only case i tried because it was the only case i needed: to get from my account images by hashtag. I havent tried anything else.

**Comment by ronilaukkarinen** [comment]
@uhlalars It worked some time with cached plugins/scripts or with access tokens that had sufficient rate limits or permissions in hand, but since Instagram won't allow hashtag feeds with these methods, it will not last. 

I see no other choice than to use a hacky scraper with hashtag feeds.
Instafeed.js is currently functional with personal user images only that doesn't need any special permissions or approval.

With scraper like [this](https://github.com/stevenschobert/instafeed.js/issues/408#issuecomment-224282075), the "good" thing is that Instagram is using JS front end and printing out the json file on their site, so it's not going to change much unless they make radical move from SPA to something else or radically change their features.

**Comment by despecial** [comment]
https://gist.github.com/mojaray2k/b502d75dd7a9fa277876

**Comment by bloqhead** [comment]
@despecial this worked for me. I was simply trying to get a user feed for a client's account and even that was giving me trouble. Let's hope it lasts. Thanks!

**Comment by kamstutz** [comment]
@benrellick Your http://instagram.pixelunion.net method for generating a valid access token worked for me on one of our sites that use instafeed.  - http://www.daytonainternationalspeedway.com   
Many thanks for that tip.  Much appreciated.

**Comment by dreshadwilliams** [comment]
The http://instagram.pixelunion.net access token worked for me as well! However they aren't "non-square" anymore. Hope I can find a solution for that soon.

**Comment by stevehanson** [comment]
@despecial that worked for me for loading a feed of my own account. It won't let me load other users' accounts, though. After generating a new access token using the [gist you shared](https://gist.github.com/mojaray2k/b502d75dd7a9fa277876), I can load my user feed with:

```javascript
var feed = new Instafeed({
  get: 'user',
  userId: '191296514',
  clientId: '191296514.xxx.xxxxxx',
  accessToken: '191296514.xxx.xxxxxx''
});
```

**Comment by waynejuckts** [comment]
@dreshadwilliams 
could you share your configuration? thats not working for me. Is public hashtag for images working for someone?

**Comment by waynejuckts** [comment]
@despecial 
with your link i got: 

Error from Instagram: This request requires scope=public_content, but this access token is not authorized with this scope. The user must re-authorize your application with scope=public_content to be granted this permissions.

**Comment by yomguy** [comment]
Here is a tiny django tag displaying a hashtag gallery (adapted from @Bolandish PHP one):
https://gist.github.com/yomguy/adaf032f6713e8abf55977a25f32d4f1

**Comment by dreshadwilliams** [comment]
@waynejuckts 
If you're looking to get other account feeds, I haven't tried that. I've only used my own, but here's my code:

```
var userFeed = new Instafeed({
	get: 'user', useHttp: true, 
	limit: 12, 
	resolution: 'standard_resolution', 
	userId: 1500521979, 
	accessToken: '1500521979.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 
	clientId: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' }); 
```

**Comment by szsoomro** [comment]
Hello. This code works for me; however, I want to specify the width and height of the images and also have spacing between the images. How do I do that? Please help!

**Comment by PrestaShark** [comment]
@brucenorton many thanks for the solution. Finaly works :+1: :)

**Comment by OmarElGabry** [comment]
What are the downsides of getting the hashtags through this [request](https://www.instagram.com/query/?q=ig_hashtag%28nofilter%29+%7B+media.first%2810%29+%7B+++count%2C+++nodes+%7B+++++caption%2C+++++code%2C+++comments+%7B+++++++count+++++%7D%2C+++++date%2C+++++dimensions+%7B+++++++height%2C+++++++width+++++%7D%2C+++++display_src%2C+++++id%2C+++++is_video%2C+++++likes+%7B+++++++count+++++%7D%2C+++++owner+%7B+++++++id%2C+++++++username%2C+++++++full_name%2C+++++++profile_pic_url%2C+++++biography+++++%7D%2C+++++thumbnail_src%2C+++++video_views%2C+++++video_url+++%7D%2C+++page_info+%7D++%7D&ref=tags%3A%3Ashow)?

**Comment by wlittler** [comment]
So maybe this is a dumb question, but how do I protect the access token? If I need to include it in my script somewhere, how can I prevent someone from viewing the source / inspecting with dev tools and copying the access token?

**Comment by bluesix** [comment]
@wlittler Tokens are locked to the userID. They can't be re-purposed for a different account. A token can only view the feed associated with its ID.

**Comment by wlittler** [comment]
The reason I ask is a few people have said "Protect this token like you
would your password", so I am concerned about it being relatively easy to
find with the appropriate knowledge.

**Comment by coolwebs** [comment]
If I am just using Instafeed to display an image "widget" tile in a homepage grid, that gets susbstantial traffic - will it keep working in Sandbox mode? I used the pixelunion method above to generate the token a few months ago and fortunately it is still working OK. I have created the app on the same account where the images are being requested. 

My big concern is the API rate limit. This web page is still in production and gets very light traffic (just our dev team). I am really worried about what will happen when it goes live and gets more than 500 requests per hour. Obviously I cannot submit it for review to make it go live as Instagram has pretty much walled that off for every scenario that I want to use this app for.

**Comment by miseixas** [comment]
Does anyone have an option for a Follow Button? The one I always use (instafollowbutton.com) is broken too :( Can't find anything working...

**Comment by l-landier** [comment]
Hello,
the instafeed is working for me, but it's publishing every post twice.
Wich means if I do limit:"4", I'll get 8 post, with every post in double (4 different images twice).

Anyone knows why?

**Comment by bluesix** [comment]
@l-landier my guess is you're calling the function twice. Post your source or URL.

**Comment by CruelMoney** [comment]
Anyone experienced that their token expired again? I've not found a way to programmatically generate tokens without redirecting to a login page. 
It is not very ideal that clients have to generate new tokens now and then when they see the feed is broken.
Any solutions to pulling client feeds without having them to generate tokens or having the users login?

**Comment by benweltman** [comment]
For anyone still experiencing issues with their instafeed - **as of Sept/ Oct 2016**

I have just fixed mine by following a mixture of the advice presented above, but I found it difficult to work out which bits to apply and for what result etc. So to help anyone else out there, here's what I did/ the requirements to get a working insta feed...

**User based instagram feed** - for those who only want a feed of one user's photos:
- You need userId, clientId and accessToken.
- Get your userId here: https://smashballoon.com/instagram-feed/find-instagram-user-id/
- Get your clientId by logging into the user account you want to display, and go to: https://instagram.com/developer - then click on Manage Clients at the top of the page, create a new client and enter app name (website name works here), any description will do, then the website url etc, but what's VIP here is go to the security tab and uncheck 'Disable Implicit OAuth', and then finally register, that's when you'll get your client id.
- Finally get your access token from here: http://instagramwordpress.rafsegat.com/docs/get-access-token/
The link above will ask for permission to access/ authorize from your profile, just do it, it's safe.

Then once you have those data items, your code should look something like:
`// Instafeed
    	var feed = new Instafeed({ 
    		get: 'user',
    		userId: '123456789',     		
    		clientId: 'f7kg85kf96u6jf85l69483k',    		    		 
    		accessToken: '123456789.f66ed08.48613c13235740a5bf5618e0b9652d35', 
    		sortBy: 'most-recent', 
    		limit: '100' 
    	}); 
    	feed.run(); `

I have of course swapped out my actual userId, clientId and accessToken but you should get the gist now.

With jquery and instafeed script initialized in the header (of html doc etc), this code and process worked for me with no issues.

Also, worth mentioning that once you have a correct access token, you can use the same one for a hashtag based feed, you just need to change your code and actually a hashtag feed requires less from you...

**Hashtag based instagram feed** - for those who want a feed of photos based on a hastag:
- You ONLY NEED the access token - which hopefully you can get from here: http://instagramwordpress.rafsegat.com/docs/get-access-token/

And for this type of feed your code should look something like:
`// Instafeed
    	var feed = new Instafeed({ 
    		get: 'tagged',
    		tagName: 'twiglets',
    		accessToken: '123456789.f66ed08.48613c13235740a5bf5618e0b9652d35', 
    		sortBy: 'most-recent', 
    		limit: '21' 
    	}); 
    	feed.run();
`

To get your access token for either of these scenarios, only use this link:
http://instagramwordpress.rafsegat.com/docs/get-access-token/ 

This is because this site adds the VERY important public_content scope as a parameter in the url for the access token, which is essentially what allows you to display hashtag content as it is considered an iffy subject under copyright and so on. YOU DO NOT NEED TO GET APPROVAL from instagram. No matter what the others have said above.

**Comment by surjithctly** [comment]
I have fixed the `This client has not been approved to access this resource` Issue by Authorizing the Instagram `public_content` scope by visiting the following URL and clicking Authorize.

`https://api.instagram.com/oauth/authorize/?client_id=[YOUR_CLIENT_ID]&redirect_uri=[YOUR_REDIRECT_URI]&scope=public_content&response_type=token`

Hope this helps someone else.

~Surjith

**Comment by mikedzl** [comment]
surjithctly - I did the same and got a "{"error_message": "Implicit authentication is disabled", "code": 403, "error_type": "OAuthForbiddenException"}" 

I tried another client id I had generated for this client and did get the "authorize" button but the page it redirected too after that said "Sorry, this page isn't available." This was the resulting URL without my specific info => https://www.instagram.com/oauth/authorize/?client_id=CLIENTID&redirect_uri=IP_OF_TEST_SERVER&response_type=code&scope=basic+public_content+follower_list+comments+relationships+likes

Has anyone discovered a way around this?

**Comment by amirhaman** [comment]
**Getting tagged insta post.**
**result:** Error from Instagram: This client has not been approved to access this resource.

I'm using following code to the the instafeed:

<code>
var feed = new Instafeed({
    get: 'tagged',
    tagName: 'TAG_NAME',
    userId: 'MY_USER_ID',
    accessToken: 'ACCESS_CODE_GENERATED_BY_ http://instagram.pixelunion.net/'
    
    
});
feed.run();
</code>

I need to get only posts which having " #TAG_NAME "

**More info:**

I can manage to get all the posts if I change get value to "user" and it will populate all the images.

any solutions to get only the tagged images?

**Comment by torfh** [comment]
Thanks it worked for me - you cannot use the accessToken provided at instagram developer site, you must go to this site : http://instagramwordpress.rafsegat.com/docs/get-access-token/. and at the buttom click "generate your access token"
Then put the new access token into your code.
I am making a website and the feed is wrapped into a html div tag.
Heres what my code looks like:

<div id="instafeed" class="tabcontent">
</div>
<script type="text/javascript">
	var userFeed = new Instafeed({
	get: 'tagged',
	tagName: 'flam****t',
	accessToken: '1909240825.ba4c844.323da3b77**********************',
	sortBy: 'most-recent',
	limit: '21' });
	userFeed.run();
	</script>
</div>

This was really quite a journey to implement this, I almost gave up when I saw the comments about "it cannot be done, it only lists your users pics with the given hashtag"

**Comment by danemacmillan** [comment]
Don't give access to some third-party site. You can generate your own access token.

http://www.logicalfeed.com/embedding-instagram-feed-in-your-website

**Comment by clintonqwert** [comment]
@rorysmorris  
Thanks! This still works!
http://instagramwordpress.rafsegat.com/docs/get-access-token/ 
the link above grants permission for both the **basic** and **public_content** scope.

ACCESS TOKEN = http://instagramwordpress.rafsegat.com/docs/get-access-token/ 
var userFeed = new Instafeed({
        get: 'tagged',  // also works with location + locationId
        tagName: 'TAGNAME',
        accessToken: 'ACCESS_TOKEN'
});
userFeed.run();

**Comment by benjamin-hull** [comment]
@clintonqwert - That's interesting! Obviously an app that got through before IG tightened everything up. Be aware (if you're not already) **public_content is going away on [December 11, 2018](https://developers.facebook.com/blog/post/2018/01/30/instagram-graph-api-updates/)** , leaving only basic scope until 2020.

@danemacmillan - the issue with self-generating tokens from your own 'app' is that you'll be permanently in basic/sandbox mode, which limits what media you can access. @rorysmorris link will give you a public_content token which will work for any user (until they turn it off, anyway - up to you whether it's worth investing in for a few months access).
